### DISCUSSION TITLE
/.vscode/sftp.json upload in environment production?

### DISCUSSION THREAD

**Discussion by morojosa** [Natizyskunk/vscode-sftp#482] [General]
I have observed that some malicious attackers try to obtain this file in production environments to obtain the keys stored there

**Comment by AureliusSlawinski** [comment]
Yes, this happened to us. We had a .htaccess inside the .vscode folder. But someone deleted the complete .vscode folder on the remote server (to clean up files). Then someone uploded the sftp.json accidentally, and without the .htaccess file the sftp.json was public (together with the ftp password).

Please allow a setting in sftp.json to specify a path to a username-password.json (or to a second sftp.json), so that this file can be located outside of any project folder. This way the file will not be uploaded in any case - even with other FTP programs that might also be used!

**Comment by Natizyskunk** [comment]
Hello @morojosa @AureliusSlawinski,
I'm aware of this problem and i'm working on a fix for it.
It should be available in the next release or so.

**Reply by Natizyskunk** [reply]
Right now I recommend using an SSH private Key file instead.

`sftp.json` file :
```
{
    [...]
    "protocol": "sftp",
    "port": 22,
    "secure": true,
    "sshConfigPath": "~/.ssh/config",
    "privateKeyPath": "~/.ssh/mywebsite.com/id_rsa"
}
```

`.ssh/config` file :
```
Host mywebsite.com 192.100.100.1
    HostName mywebsite.com
    User username
    PreferredAuthentications publickey
    IdentityFile ~/.ssh/mywebsite.com/id_rsa
 ```

**Reply by AureliusSlawinski** [reply]
Thanks for the info!
